### DISCUSSION TITLE
Enable search and filter by metadata

### DISCUSSION THREAD

**Discussion by mffap** [zitadel/zitadel#2453] [Ideas]
Users should be able to search and filter by user and org metadata.

For filter, we could provide an option to add more columns (based on the metadata). Maybe the allowed set of 'keys' to filter could be defined to avoid a huge mass of keys in the selection.

User metadata
![image](https://user-images.githubusercontent.com/1366906/135602937-41880348-313b-4431-a465-9afa1eef5bd9.png)

Org metadata
![image](https://user-images.githubusercontent.com/1366906/135603152-44358da9-f4d5-46ef-bdd5-60523cf1a3f1.png)

**Comment by ghstahl** [comment]
A lucene type query?

**Reply by fforootd** [reply]
that would be cool, yeah

**Comment by ahmednfwela** [comment]
Just hit this recently, we need to be able to use the metadata as if it were a normal property, that means filtering + searching and including the metadata in the User object itself, rather than having a separate request for it (or the ability to request multiple user metadata at once).
